dashboard/test/ui/runner.rb

```diff
@@ -38,187 +39,246 @@
-    $options.retry_count = times_to_retry.to_i
-  end
-  opts.on("--magic_retry", "Magically retry tests based on how flaky they are") do
-    $options.magic_retry = true
-  end
-  opts.on("--abort_when_failures_exceed Limit", Numeric, "Maximum allowed feature failures before the whole test run is aborted (default is infinity)") do |max_failures|
-    $options.abort_when_failures_exceed = max_failures
-  end
-  opts.on("-n", "--parallel ParallelLimit", String, "Maximum number of browsers to run in parallel (default is 1)") do |p|
-    $options.parallel_limit = p.to_i
-  end
-  opts.on("--db", String, "Run scripts requiring DB access regardless of environment (otherwise restricted to development/test).") do
-    $options.force_db_access = true
-  end
-  opts.on("-V", "--verbose", "Verbose") do
-    $options.verbose = true
-  end
-  opts.on("--fail_fast", "Fail a feature as soon as a scenario fails") do
-    $options.fail_fast = true
-  end
-  opts.on('-s', '--script Scriptname', String, 'Run tests associated with this script, or have Scriptname somewhere in the URL') do |scriptname|
-    f = `egrep -r "Given I am on .*#{scriptname.delete(' ').downcase}" . | cut -f1 -d ':' | sort | uniq | tr '\n' ,`
-    $options.feature = f.split ','
-  end
-  opts.on('--with-status-page', 'Generate a test status summary page for this test run') do
-    $options.with_status_page = true
-    $options.html = true # Implied by wanting a status page
-  end
-  opts.on('--output-synopsis', 'Print a synopsis of failing scenarios') do
-    $options.output_synopsis = true
-  end
-  opts.on_tail("-h", "--help", "Show this message") do
-    puts opts
-    exit
   end
 end
 
-opt_parser.parse!(ARGV)
-passed_features = ARGV + ($options.feature || [])
-# Standardize: Drop leading dot-slash on feature paths
-passed_features.map! {|feature| feature.gsub(/^\.\//, '')}
-
-$browsers = JSON.load(open("browsers.json"))
-
-$lock = Mutex.new
-$suite_start_time = Time.now
-$suite_success_count = 0
-$suite_fail_count = 0
-# How many flaky test reruns occurred across all tests (ignoring the initial attempt).
-$total_flaky_reruns = 0
-$total_flaky_successful_reruns = 0
-$failures = []
-
-if $options.local
-  SeleniumBrowser.ensure_chromedriver_running
-  $browsers = [{
-    "browser": "local",
-    "name": "ChromeDriver",
-    "browserName": "chrome",
-    "version": "latest"
-  }]
-end
-
-if $options.config
-  $browsers = $options.config.map do |name|
-    $browsers.detect {|b| b['name'] == name}.tap do |browser|
-      unless browser
-        puts "No config exists with name #{name}"
-        exit
+def select_browser_configs(options)
+  if options.local
+    SeleniumBrowser.ensure_chromedriver_running
+    return [{
+      'browser': 'local',
+      'name': 'ChromeDriver',
+      'browserName': 'chrome',
+      'version': 'latest'
+    }]
+  end
+
+  browsers = JSON.load(open("browsers.json"))
```

Comment:
Can this be `JSON.parse` rather than `JSON.load`?
https://github.com/bbatsov/rubocop/blob/master/manual/cops_security.md#securityjsonload
http://ruby-doc.org/stdlib-2.3.0/libdoc/json/rdoc/JSON.html#method-i-load